quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java

```diff
@@ -51,6 +53,17 @@ public static PropertyMapper<?>[] getDatabasePropertyMappers() {
                         .mapFrom(DatabaseOptions.DB, DatabasePropertyMappers::getDatabaseUrl)
                         .paramLabel("jdbc-url")
                         .build(),
+                fromOption(DatabaseOptions.DB_POSTGRESQL_TARGET_SERVER_TYPE)
+                        .to("quarkus.datasource.jdbc.additional-jdbc-properties.targetServerType")
+                        .mapFrom(DatabaseOptions.DB, (db, c) ->
+                                getPostgresqlTargetServerType(
+                                        db,
+                                        c))
+                        .isEnabled(() ->
+                                getPostgresqlTargetServerType(
+                                        Configuration.getConfigValue(DB).getValue(),
+                                        null) != null)
```

Comment:
```suggestion
                        .mapFrom(DatabaseOptions.DB, DatabasePropertyMappers::getPostgresqlTargetServerType)
                        .isEnabled(() -> getPostgresqlTargetServerType(Configuration.getConfigValue(DB).getValue(), null) != null)
```

Would be nicer to have it like this if you don't mind.